.github/scripts/check-via-structural-rules.sh

```
@@ -0,0 +1,81 @@
+SGCONFIG="$AST_GREP_DIR/sgconfig.yml"
+
+if [[ ! -f "$SGCONFIG" ]]; then
+    echo "Error: ast-grep config not found at $SGCONFIG" >&2
+    exit 1
+fi
+
+set +e
+# Run ast-grep scan from repo root.
+# Use --color=never for robust parsing of the output in CI/scripts.
+OUTPUT=$(cd "$REPO_ROOT" && ast-grep scan --config "$SGCONFIG" --color=never 2>&1)
+EXIT_CODE=$?
```

Shellcheck (0.11.0)

[warning] EXIT_CODE appears unused. Verify use (or export if used externally).

(SC2034)